drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb

```diff
@@ -0,0 +1,525 @@
+    ANSWER = 'Answer'.freeze
+
+    def perform
+      rows = @reader.rows(filename: FILENAME, field_id_row_number: nil)
+      clobber_records(rows) if clobber
+
+      @services = create_services(rows)
+      ar_import(service_class, @services.values) # We need to save these first as we need the ids for CDEs
+
+      @cdes = create_records(rows)
+      ar_import(cde_class, @cdes)
+    end
+
+    private def clobber_records(rows)
+      custom_service_ids = [].tap do |ids|
+        rows.each do |row|
+          row_field_value = row.field_value(TOUCHPOINT_NAME)
+          config = configs[row_field_value]
+          next if config.blank?
+
+          ids << generate_service_id(config, row)
+        end
+      end
+      services = service_class.where(CustomServiceID: custom_service_ids, data_source: data_source.id)
+
+      rails_service_ids = [].tap do |ids|
+        services.find_each do |service|
+          ids << service.id
+        end
+      end
+
+      services.delete_all
+
+      cdes = cde_class.where(owner_type: service_class.name, owner_id: rails_service_ids)
+      cdes.delete_all
+    end
+
+    private def create_services(rows)
+      log_info('creating services')
+      expected = 0
+      actual = 0
+
+      enrollments = Hmis::Hud::Enrollment.where(data_source_id: data_source.id).index_by(:enrollment_id)
+      service_type_ids = Hmis::Hud::CustomServiceType.where(data_source_id: data_source.id).pluck(:name, :id).to_h
+
+      result = {}.tap do |services|
+        rows.each do |row|
+          expected += 1
+
+          row_field_value = row.field_value(TOUCHPOINT_NAME)
+          config = configs[row_field_value]
+          if config.blank?
+            log_skipped_row(row, field: TOUCHPOINT_NAME)
+            next
+          end
+
+          response_id = row.field_value(RESPONSE_ID)
+
+          service_type = config[:service_type]
+          service_type_id = service_type_ids[service_type]
+          if service_type_id.blank?
+            log_info("Service type configuration error: can't find #{service_type}!")
+            log_skipped_row(row, field: :service_type)
+            next
+          end
+
+          row_field_value = row.field_value(ENROLLMENT_ID)
+          enrollment = enrollments[row_field_value]
+          if enrollment.blank?
+            log_skipped_row(row, field: ENROLLMENT_ID)
+            next
+          end
+
+          actual += 1
+
+          services[response_id] ||= service_class.new(
+            CustomServiceID: generate_service_id(config, row),
+            EnrollmentID: enrollment.EnrollmentID,
+            PersonalID: enrollment.PersonalID,
+            UserID: system_hud_user.id,
+            DateProvided: parse_date(row.field_value(DATE_PROVIDED)),
+            data_source_id: data_source.id,
+            custom_service_type_id: service_type_id,
+            service_name: service_type,
+            DateCreated: today,
+            DateUpdated: today,
+            FAAmount: nil,
```

Comment:
Services that collect financial assistance amount should store it in this field, _not_ as a custom data element. I think here you're storing it in data elements with key "FAAmount", which we don't want

Reply:
nevermind, I see they are nested under `service_fields`